stub-idp/src/main/java/uk/gov/ida/stub/idp/repositories/StubCountryRepository.java

```diff
@@ -33,4 +44,21 @@ public StubCountry getStubCountryWithFriendlyId(EidasScheme eidasScheme){
                 allIdpsUserRepository
         );
     }
+
+    public List<UserCredentials> getUserCredentialsForFriendlyId(String friendlyId) {
+        final Optional<IdpStubsConfiguration> idpStubsConfiguration = idpStubsConfigurationFactory.tryBuildDefault();
+        if (!idpStubsConfiguration.isPresent()) {
+            return Collections.emptyList();
+        }
+
+        final List<StubIdp> matchingCountries = idpStubsConfiguration.get().getStubCountries().stream()
+                .filter(c -> c.getFriendlyId().equals(friendlyId))
+                .collect(Collectors.toList());
+
+        if (matchingCountries.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        return matchingCountries.get(0).getIdpUserCredentials();
```

Comment:
This will fail in an unhelpful way as soon as we add another country. Obviously we have no plans to do that right now but who knows when down the line it might happen.

Can't we just loop over the `matchingCountries` and add each set of user credentials to a list? That way we don't even need the `if`-statement.

Reply:
stub-country can already pretend to be other countries right now without any additional configuration, but [before this PR] all users come from stub-country

e.g. you can configure http://stub-idp/france/ServiceMetadata in fedconfig/metadata and it will just work without additional configuration in the stub-country app (so there could be a test environment with all countries listed if it was wanted)

Reply:
https://github.com/alphagov/verify-stub-idp/blob/2d58bcd5a675c78257cd6c43b52005df9da3ca23/stub-idp/src/main/java/uk/gov/ida/stub/idp/domain/EidasScheme.java#L9-L42